src/external/wannier_shared.jl

```diff
@@ -0,0 +1,407 @@
+# Shared functionality to operate with Wannier programs (Wannier.jl and Wannier90).
+# The communication happens with the Wannier90 file formats which Wannier.jl also understands.
+
+using Dates
+
+@doc raw"""
+Base type for Wannier projections.
+
+Wannierization searchs for a unitary matrix ``U_{m_n}``.
+As a starting point for the search, we can provide an initial guess function ``g``
+for the shape of the Wannier functions, based on what we expect from knowledge of the problem or physical intuition.
+This starting matrix is called ``[A_k]_{m,n}``, and is computed as follows:
+``[A_k]_{m,n} = \langle ψ_m^k | g^{\text{per}}_n \rangle``.
+The matrix will be orthonormalized by the chosen Wannier program, we don't need to do so ourselves.
+"""
+abstract type WannierProjection end
+
+# Dispatch function for WannierProjection implementations, evaluating g in Fourier space.
+function get_fourier_projection_coefficients end
```

Comment:
Could simply be the call operator, i.e.
```
function (proj::GaussianWannierProjection)(basis::PlaneWaveBasis, qs)
   # stuff
end
```
That makes it even more flexible as one could pass arbitrary functions instead of these predefined functors when doing the wannierisation.